fmriprep/workflows/fieldmap/base.py

```diff
@@ -36,48 +36,223 @@
+                     for fmap_ in fmaps if fmap_['type'] == 'epi']
+        sdc_unwarp_wf = init_pepolar_unwarp_wf(
+            bold_meta=bold_meta,
+            epi_fmaps=epi_fmaps,
+            omp_nthreads=omp_nthreads,
+            name='pepolar_unwarp_wf')
+
+    # FIELDMAP path
+    if fmap['type'] in ['fieldmap', 'phasediff']:
+        setattr(workflow, 'sdc_method', 'FMB (%s-based)' % fmap['type'])
+        # Import specific workflows here, so we don't break everything with one
+        # unused workflow.
+        if fmap['type'] == 'fieldmap':
+            from .fmap import init_fmap_wf
+            fmap_estimator_wf = init_fmap_wf(
+                omp_nthreads=omp_nthreads,
+                fmap_bspline=fmap_bspline)
+            # set inputs
+            fmap_estimator_wf.inputs.inputnode.fieldmap = fmap['fieldmap']
+            fmap_estimator_wf.inputs.inputnode.magnitude = fmap['magnitude']
+
+        if fmap['type'] == 'phasediff':
+            from .phdiff import init_phdiff_wf
+            fmap_estimator_wf = init_phdiff_wf(omp_nthreads=omp_nthreads)
+            # set inputs
+            fmap_estimator_wf.inputs.inputnode.phasediff = fmap['phasediff']
+            fmap_estimator_wf.inputs.inputnode.magnitude = [
+                fmap_ for key, fmap_ in sorted(fmap.items())
+                if key.startswith("magnitude")
+            ]
+
+        sdc_unwarp_wf = init_sdc_unwarp_wf(
+            omp_nthreads=omp_nthreads,
+            fmap_demean=fmap_demean,
+            debug=debug,
+            name='sdc_unwarp_wf')
+
+        workflow.connect([
+            (inputnode, sdc_unwarp_wf, [
+                ('name_source', 'inputnode.name_source'),
+                ('bold_ref', 'inputnode.in_reference'),
+                ('bold_ref_brain', 'inputnode.in_reference_brain'),
+                ('bold_mask', 'inputnode.in_mask')]),
+            (fmap_estimator_wf, sdc_unwarp_wf, [
+                ('outputnode.fmap', 'inputnode.fmap'),
+                ('outputnode.fmap_ref', 'inputnode.fmap_ref'),
+                ('outputnode.fmap_mask', 'inputnode.fmap_mask')]),
+        ])
+
+    # FIELDMAP-less path
+    if fmaps[-1]['type'] == 'syn':
+        syn_sdc_wf = init_syn_sdc_wf(
+            bold_pe=bold_meta.get('PhaseEncodingDirection', None),
+            omp_nthreads=omp_nthreads)
+
+        workflow.connect([
+            (inputnode, syn_sdc_wf, [
+                ('t1_brain', 'inputnode.t1_brain'),
+                ('t1_2_mni_reverse_transform', 'inputnode.t1_2_mni_reverse_transform'),
+                ('bold_ref', 'inputnode.bold_ref'),
+                ('bold_ref_brain', 'inputnode.bold_ref_brain'),
+                ('template', 'inputnode.template')]),
+        ])
+
+        # XXX Eliminate branch when forcing isn't an option
+        if len(fmaps) == 1:  # --force-syn was called
+            setattr(workflow, 'sdc_method', 'FLB ("fieldmap-less" based) - SyN')
```

Comment:
```Python
workflow.sdc_method = '...'
```

Also, I don't know that "fieldmap-less based" makes sense. Maybe "fieldmap-less, SyN-based", or just "fieldmap-less".